app/javascript/components/time_table/timesheet/entry.js

```diff
@@ -160,17 +175,22 @@ class Entry extends React.Component {
   }
 
   formattedHoursAndMinutes (time) {
-    let hours = moment(time).hours();
-    let minutes = moment(time).minutes();
-    if (hours < 10) hours = `0${hours}`;
-    if (minutes < 10) minutes = `0${minutes}`;
+    return moment(time).format('HH:mm');
+  }
+
+  inclusiveParse(time) {
+    const firstFormat = moment(time, 'HH:mm');
+    if (firstFormat.isValid()) {
+      return firstFormat;
+    }
 
-    return `${hours}:${minutes}`;
+    // Format for f('835') = 08:35
```

Comment:
lets make this comment more clear: i.e. Properly handle various time inputs: 0830 -> 08:30, 830 -> 08:30 etc